main.cpp

```diff
@@ -64,151 +63,592 @@ New/This/Pointers/References conclusion
+    *value = *value / *dt.value;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+// DoubleType member functions implementation
+// IntType argument
+DoubleType& DoubleType::add(const IntType& it)
+{
+    std::cout << "(dt) " << *value << " + (it) " << *it.value << " = ";
+    *value = *value + *it.value;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+DoubleType& DoubleType::subtract(const IntType& it)
+{
+    std::cout << "(dt) " << *value << " - (it) " << *it.value << " = ";
+    *value = *value - *it.value;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+DoubleType& DoubleType::multiply(const IntType& it)
+{
+    std::cout << "(dt) " << *value << " * (it) " << *it.value << " = ";
+    *value = *value * *it.value;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+DoubleType& DoubleType::divide(const IntType& it)
+{
+    if (*it.value == 0)
+    {
+        std::cout << "Error: cannot divide by 0!\n";
+    }
+    std::cout << "(dt) " << *value << " / (it) " << *it.value << " = ";
+    *value = *value / *it.value;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+// IntType member functions implementation
+// Primitives
+IntType& IntType::add(int v)
+{
+    std::cout << "(it) " << *value << " + " << v << " = ";
+    *value += v;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+IntType& IntType::subtract(int v)
+{
+    std::cout << "(it) " << *value << " - " << v << " = ";
+    *value -= v;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+IntType& IntType::multiply(int v)
+{
+    std::cout << "(it) " << *value << " * " << v << " = ";
+    *value *= v;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+IntType& IntType::divide(int v)
+{
+    if (v == 0.0f)
+    {
+        std::cout << "Error: cannot divide by 0!\n";
+    }
+    std::cout << "(it) " << *value << " / " << v << " = ";
+    *value /= v;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+// IntType member functions implementation
+// FloatType argument
+IntType& IntType::add(const FloatType& ft)
+{
+    std::cout << "(it) " << *value << " + (ft) " << *ft.value << " = ";
+    *value = *value + *ft.value;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+IntType& IntType::subtract(const FloatType& ft)
+{
+    std::cout << "(it) " << *value << " - (ft) " << *ft.value << " = ";
+    *value = *value - *ft.value;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+IntType& IntType::multiply(const FloatType& ft)
+{
+    std::cout << "(it) " << *value << " * (ft) " << *ft.value << " = ";
+    *value = *value * *ft.value;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+IntType& IntType::divide(const FloatType& ft)
+{
+    if (*ft.value == 0)
+    {
+        std::cout << "Error: cannot divide by 0!\n";
+    }
+    std::cout << "(it) " << *value << " / (ft) " << *ft.value << " = ";
+    *value = *value / *ft.value;
+    std::cout << *value << '\n';
```

Comment:
I could be mistaken, but I believe you might get warnings about this line and the implicit conversion.   You'll have to try feeding a `FloatType(0)` to this function and see if you get a crash.

Reply:
This is the result:
`(it) 1 / (ft) 0 = -2147483648`
I'll skip the calculation in this case as well.

I have suppressed 3 warning flags in .replit though:
`-Wno-float-equal -Wno-conversion -Wno-double-promotion`
Am i actually not supposed to have done this?